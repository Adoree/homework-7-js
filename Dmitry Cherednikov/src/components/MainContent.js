import React from 'react';
import { Switch, Route } from 'react-router-dom';

import CatchedList from '../containers/CatchedList';
import PokemonsList from "../containers/PokemonsList";
import PokePage from "../containers/PokePage";

const MainContent = () => (
  <main>
    <Switch>
      <Route exact path='/pokemons' component={PokemonsList}/>
      <Route exact path='/pokemons/:id' component={PokePage}/>
      <Route exact path='/catched' component={CatchedList}/>
    </Switch>
  </main>
);

export default MainContent;