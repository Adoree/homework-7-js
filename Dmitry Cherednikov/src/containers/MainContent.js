import React, { Component } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import { fetchPokemons, fetchCatchedPokemons } from '../actions/index';
import { getError, getIsFetchingPokemons, getIsFetchingCatched, getPokemons, getCatchedPokemons } from '../reducers/index';
import CatchedList from './CatchedList';
import Content from '../components/Content';
import FetchError from '../components/FetchError'

class MainContent extends Component {
  static propTypes = {
    isFetchingPokemons: PropTypes.bool.isRequired,
    isFetchingCatched: PropTypes.bool.isRequired,
    pokemonList: PropTypes.array.isRequired,
    catchedPokemons: PropTypes.array.isRequired,
  };

  static defaultProps = {
    fetchPokemons: () => {},
    fetchCatchedPokemons: () => {},
  };

  componentDidMount() {
    this.fetch();
  }

  fetch = () => {
    const { fetchPokemons, fetchCatchedPokemons } = this.props;
    fetchPokemons();
    fetchCatchedPokemons();
  };

  render() {
    const { isFetchingPokemons, isFetchingCatched, pokemonList, catchedPokemons, errorMessage } = this.props;

    if (isFetchingPokemons && isFetchingCatched) {
      return (
        <p>Loading...</p>
      )
    }

    if (errorMessage && !pokemonList.length && !catchedPokemons) {
      return (
        <FetchError
          message={errorMessage}
          onRetry={this.fetch}
        />
      )
    }

    return (
      <main>
        <Switch>
          <Route path='/pokemons' component={Content}/>
          <Route path='/catched' component={CatchedList}/>
        </Switch>
      </main>
    )
  }
}

const mapStateToProps = (state) => ({
  isFetchingPokemons: getIsFetchingPokemons(state),
  isFetchingCatched: getIsFetchingCatched(state),
  pokemonList: getPokemons(state),
  catchedPokemons: getCatchedPokemons(state),
  errorMessage: getError(state),
});

const mapDispatchToProps = (dispatch) => ({
  fetchPokemons: bindActionCreators(fetchPokemons, dispatch),
  fetchCatchedPokemons: bindActionCreators(fetchCatchedPokemons, dispatch),
});

MainContent = withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContent));

export default MainContent;