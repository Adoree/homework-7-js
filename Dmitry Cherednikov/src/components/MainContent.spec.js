import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainContent from './MainContent';

describe('<MainContent />', () => {
  it('should render correctly', () => {
    const wrapper = shallow((
      <MainContent />
    ));

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
