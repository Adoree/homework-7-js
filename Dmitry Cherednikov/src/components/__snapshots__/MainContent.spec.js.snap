// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContent /> should render correctly 1`] = `
<main>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/pokemons"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/pokemons/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/catched"
    />
  </Switch>
</main>
`;
